driver-core/src/main/com/mongodb/internal/connection/DefaultClusterFactory.java

```diff
@@ -70,26 +74,33 @@ public Cluster createCluster(final ClusterSettings clusterSettings, final Server
 
         DnsSrvRecordMonitorFactory dnsSrvRecordMonitorFactory = new DefaultDnsSrvRecordMonitorFactory(clusterId, serverSettings);
 
+        AsynchronousClusterEventListener clusterEventListener =
```

Comment:
This is where the rubber meets the road: we wrap the listeners from cluster and server settings into a single listener.